src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -52,567 +51,718 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
-            Set<Id> relatedAccountIds = new Set<Id>();
-            List<Account> relatedAccounts = new List<Account>();
-            for (Contact c : oldContacts) {
-                listContactAccountDelete.add(c);
-                //Collect related account ids
-                relatedAccountIds.add(c.AccountId);
+            //Account needs to be created if (1) Account does not exist or (2) Account model is not blank
+            if (c.AccountId == NULL && defaultRecTypeId != NULL) {
+                contactsNeedAccounts.add(c);
             }
 
-            //Re-query to get correct Account values
-            relatedAccounts = [SELECT Id,
-                                    RecordTypeId
-                                FROM Account
-                                WHERE Id IN :relatedAccountIds];
-            for (Account account : relatedAccounts) {
-                // Rename the account if a contact is deleted from a household account
-                if (checkHouseholdNaming(account.RecordTypeId)) {
-                    setAccountIdHHToRename.add(account.Id);
+            //AFTER INSERT
+            if (triggerAction == TDTM_Runnable.Action.AfterInsert &&
+                !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Insert)) {
+                    insertWrapper = new HandleInsertWrapperLogic (c);
+                }
+
+            //AFTER UPDATE
+            if (triggerAction == TDTM_Runnable.Action.AfterUpdate &&
+                !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Update)) {
+                    updateWrapper = new HandleUpdateWrapperLogic (c, oldContacts, triggerAction, i);
+                }
+            i += 1;
+        }
+
+        //Set recursion flag
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Insert, true);
+        }
+
+        if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Update, true);
+        }
+
+        if (triggerAction == TDTM_Runnable.Action.AfterDelete) {
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Delete, true);
+        }
+
+        handleInsertProcessing(insertWrapper, contactsNeedAccounts);
+        handleUpdateProcessing(updateWrapper, contactsNeedAccounts);
+    }
+
+    /*******************************************************************************************************
+    * @description Processes after insert logic.
+    * @return HandleInsertWrapperLogic.
```

Comment:
Not necessary.

Reply:
@robertbwatson , I've removed the return. It's fine if I keep the description right?